### PR TITLE
Set waku mode dynamically

### PR DESCRIPTION
**Pull request by cammellos**
This commit allows setting waku-mode and waku-bloom-filter-mode
dynamically.
It requires a relogin for the changes to take effect.

status: ready

### CODE REVIEW

**Comment by Serhy** [comment]
I see it's WIP, but as e2e failed and same on real devices - offline message history doesn't work with `eth.prod` and disabled Waku setting.

When Waku enabled and i'm on `eth.test` or `eth.staging` there is endless "Connecting..." and no way to send messages in chats (no Send button)

**Comment by cammellos** [comment]
thanks, I will take a look!

File: src/status_im/multiaccounts/update/core.cljs [hunk 1]
```diff
@@ -7,10 +7,10 @@
             [status-im.utils.types :as types]
             [taoensso.timbre :as log]))
 
-(fx/defn send-multiaccount-update [{:keys [db]}]
+(fx/defn send-multiaccount-update [{:keys [db] :as cofx}]
   (let [multiaccount (:multiaccount db)
         {:keys [name preferred-name photo-path address]} multiaccount]
-    {::json-rpc/call [{:method (json-rpc/call-ext-method "sendContactUpdates")
+    {::json-rpc/call [{:method (json-rpc/call-ext-method (get-in db [:multiaccount :waku-enabled]) "sendContactUpdates")
```
Comment (rasom) [inline src/status_im/multiaccounts/update/core.cljs:13]: why not `waku/enabled?`?

**Review by rasom — APPROVED** [review]

**Comment by Serhy** [comment]
All 8 failed tests above https://github.com/status-im/status-react/pull/10092#issuecomment-592685461 (with Waku mode ON and `eth.test` fleet set by default) failed because those tests after curtain actions re-login and in that case, - after re-login due to Log Out->Login or Changing network (forced logout), - Waku toggle button became OFF, which caused endless 'Connecting...' with empty main `Chats` view.

I've reproduced the same with emulator and real device: if Waku setting enabled by default, - it will be OFF after the first relogin.
However, if disabled `Waku enabled` setting then to toggle ON in all above cases, all further relogins keep this setting in it's state. 

So it's something related to `Waku enabled`= ON setting does not being "saved" in settings on the first app start thus drops to OFF after re-login.

Manually all failed tests in re-tested and all looks fine (if I force `Waku enabled`). 

But if we are going at some point to have builds having `Waku enabled` by default - we need to fix the issue of dropping the setting after logout. 

The rest works pretty the same to whisper (`eth.test` set as fleet and `Waku` enabled):
- Can messaging between iOS and Android in 1-1 and public and Group Chat.
- Offline history received fine
- Account pairing works the same to develop
- Can send transactions in 1-1 chat (and see Confirmed state after it confirmed)
- Can messaging between different networks
- Can see and change ENS names in chats
- Can block a user and no messages are received in any chat 
- Can delete chats and clear history which is persisted as expected.

Thing which is outside of this PR is a bridge between whisper and waku. 

No issues in with messaging while Waku is enabled, but only minor issue with "Waku enabled" being toggled OFF after relogin if it was enabled by default in app.

**Comment by Serhy** [comment]
I have checked latest two builds here which was working before in `eth.test` and Waku mode fine, but something changed apparently with `eth.test` because it doesn't work anymore.

1) Install https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200301-154430-5e8e25-pr-universal.apk
2) Create new account
3) Switch account to `eth.test` fleet and relogin.
4) Toggle `Waku enabled` setting into ON state. You'll be log out as expected.
5) Login to account again. 

**Actual result:** `Waku enabled` is OFF. Endless `Connecting...` on the chats screen. Logcat says: 
```
03-05 14:17:26.629  9869  9957 E unknown:ReactNative: console.error: "ERROR [status-im.events:266] - Error on mark-trusted-peer:  {:code -32601, :message "the method shh_markTrustedPeer does not exist/is not available"}", js engine: hermes, stack:
03-05 14:17:26.629  9869  9957 E unknown:ReactNative: apply@-1
03-05 14:17:26.629  9869  9957 E unknown:ReactNative: c (address at@1:4154132
03-05 14:17:26.629  9869  9957 E unknown:ReactNative: f (address at@1:4153745
03-05 14:17:26.629  9869  9957 E unknown:ReactNative: call@-1
03-05 ........
......
......

```

Full logcat with previously set Waku in ON state: [logcat.log](https://github.com/status-im/status-react/files/4292880/Waku_not_works_logcat.log)


Protocol is set Waku though looking at logs, even `Waku enabled` is OFF as toggle setting.

**Comment by adambabik** [comment]
When waku is enabled, the app should send JSON-RPC calls with `waku_` prefix instead of `shh_`. Whisper is not started so `shh_` methods are not available. It needs to be changed to `waku_`.

**Comment by Serhy** [comment]
Alright, it's something with how `Waku enabled` setting applied 
What happens now:
1) I install latest Status build  (https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200301-154430-5e8e25-pr-universal.apk)  and create new account
2) In `Profile` -> `Advanced` there is `eth.prod` and `Waku enabled` false
3) Switch fleet to `eth.test` and relogin
**Actual result:** `Node config:{"ClusterConfig":{"Enabled":true,"Fleet":"eth.test"....`
4) In `Profile` -> `Advanced` toggle `Waku enabled` onto ON
**Actual result:** `Node config:{"ClusterConfig":{"Enabled":true,"Fleet":"eth.test".............."WakuConfig":{"Enabled":true,"BloomFilterMode":null` in config, but
5) Login to account and check `Profile` -> `Advanced`

**Actual result:** Waku toggle disabled. Endless `Connecting...` on chats view. Logs are same to above comment 
```

03-05 17:35:01.994  7095  7134 E ReactNativeJS: ERROR [status-im.mailserver.core:96] - mailserver: update-mailservers error {:code -32601, :message "the method shhext_updateMailservers does not exist/is not available"}
03-05 17:35:02.023  7095  7134 E ReactNativeJS: ERROR [status-im.transport.shh:29] - :shh/get-new-sym-key-error {:code -32601, :message "the method shh_generateSymKeyFromPassword does not exist/is not available"}
03-05 17:35:02.031  7095  7138 E unknown:ReactNative: console.error: "ERROR [status-im.mailserver.core:96] - mailserver: update-mailservers error {:code -32601, :message "the method shhext_updateMailservers does not exist/is not available"}", js engine: hermes, stack:
03-05 17:35:02.031  7095  7138 E unknown:ReactNative: apply@-1
03-05 17:35:02.031  7095  7138 E unknown:ReactNative: c (address at@1:4154132

```
**Expected result:** Waku is enabled in `Profile`

Now, if I re-open the app (or change network with forced relogin) then SHH is used as expected  (because `Waku enabled` being OFF before relogin) 

Waku worked (and works now) in 4th build https://github.com/status-im/status-react/commit/6cf0d8c714133caa5da0b4abdbf7c6941621d7c2 and I see that bridging seem to be working from the first view (I posted messages in 4th build with Waku working and checked last 6th build in `eth.test` + SHH protocol)

**Comment by Serhy** [comment]
This is how it should be based on plan we are going to deliver Waku in subsequent release (@andremedeiros  please confirm)

1) Install Status build and create new account
**Expected result = Actual result:** eth.prod and Waku disabled is set by default 

2) Switch to `eth.test` fleet and re-login to account
**Expected result:** eth.test fleet is set AS WELL AS Waku enabled along with that (so `Waku enabled` option is toggled ON)

3) Disable `Waku enabled` option. Relogin.
**Expected result:** `eth.test` fleet is set. Waku disabled (so Whisper is used now)

**Comment by andremedeiros** [comment]
> Install Status build and create new account

This should set you up with Waku automatically.

> Switch to eth.test fleet and re-login to account

This should still have Waku enabled.

> Disable Waku enabled option. Relogin.

You're correct in that Waku should be disabled and Whisper should be used.

**Comment by cammellos** [comment]
@Serhy last build was using the wrong status-go version ( I didn't merge a PR in status-go), everything should be back as it was with the last commit.

**Comment by Serhy** [comment]
Thanks @andremedeiros for explanation https://github.com/status-im/status-react/pull/10092#issuecomment-595401175

Builds are working fine now, @cammellos, thanks a lot! (tested Android only so far because iOS failed which fixed by https://github.com/status-im/status-react/commit/8b3c7f29b27827cea9537eceb30ff5978111ba61 I hope you are OK for me to rebase develop and push it)

Also I confirmed messaging works (including offline) if UserA (uses Release1.1) with `eth.test` fleet (and Whisper obviously because no Waku in next release) sends messages to UserB who uses PR build with `eth.test` and `Waku enabled`=True. This is important because we can not guarantee all users will update to Release1.2 (supposed roll out waku release version) with `Waku` while `eth.prod` will be updated along with Release1.2 rollout.

Rebasing develop to make successful iOS build...

**Comment by Serhy** [comment]
Rebased this PR branch in another PR #10145 which successfully built iOS. 
Tested iOS and Android with Whisper <-> Waku setting on eth.test fleet which looks good to me (offline message history received, messages are exchanged fine, pairing between Release and PR build works)

Messages from Release1.1 <-> PR build (with Waku-enabled) exchanged fine (when on `eth.test` fleet) 

As expected here at the moment: Waku setting is disabled by default because `eth.prod`, as default fleet, - has no Waku changes yet. And it's good!

I'd like to `eth.staging` with Waku go version and briefly test it on `eth.staging` first before making the same with eth.prod which is used by current releases. 

When we are about to release Release1.2 we'll go with:
(1) Enable `Waku enabled` setting by default
(2) Have a Waku deployed to `eth.prod`
(3) Release1.2 to markets

**Comment by andremedeiros** [comment]
> When we are about to release Release1.2 we'll go with:
(1) Enable Waku enabled setting by default
(2) Have a Waku deployed to eth.prod
(3) Release1.2 to markets

There will be other things bundled with the Waku release, but yeah.

**Pull request merged**